File: python/sorting/bubbleSort.py
def bubbleSort(arr):
    arrLen = len(arr)
    for i in range(arrLen-1):
        wasSwapped = False
        for j in range(0, arrLen-i-1):
            if arr[j] > arr [j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]
                wasSwapped = True  
        if not wasSwapped:
            break
    print("Sorted Array ", arr)

bubbleSort([4,2,5,3,1,7])

File: javascript/sorting/bubbleSort.js
/* Author : Shubhankar Kotnala
 * Problem : Sort a given array using Bubble Sort
 * Explanation : Bubble sort puts the largest element
 * at the end of the array in each iteration.
 */
function bubbleSort(arr){
    var len = arr.length;
    for(var i = 0; i<len; i++){

        //to break the control loop to increase the efficiency of the code
        var wasSwapped = false;
        for(var j=0; j< len - i; j++){
            if(arr[j] > arr[j+1]){
                var tempHolder = arr[j];
                arr[j] =arr[j+1];
                arr[j+1] = tempHolder;
                wasSwapped = true;
            }
        }
        //If swapping is not performed in the entire iteration, the array is sorted
        if(!wasSwapped)
            break;
    }

    console.log("Sorted Array ", arr);
}

bubbleSort([4,2,5,3,1,7]);